src/lib/addClipMarkup.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Add clip-related markup to page head for Google SEO.
+ *
+ * @param {player} Player An instance of the Player.
+ * @return {void}
+ */
+// TODO: NEED TO DETERMINE WHERE THIS METHOD SHOULD BE DEFINED, IF NOT HERE
+export function addClipMarkup(player) {
+    if (!player) {
+        return;
+    }
+
+    // Any off-site page may have multiple embedded Vimeo videos. We can inject
+    // VideoObject markup for each video. According to the Google SEO team, we
+    // should include 'chapters' data for only one video in the page
+    const scriptElem = document.querySelector("script[type='application/ld+json']");
+    let existingMicrodataHasChapters;
+    let existingMicrodata = [];
+
+    if (scriptElem) {
+        const temp = JSON.parse(scriptElem.textContent);
+        existingMicrodata = Array.isArray(temp) ? temp.slice() : new Array(temp);
+        existingMicrodataHasChapters = existingMicrodata.some((item) => item.hasOwnProperty('hasPart'));
+    }
+
+    player.get('videoObjectMetadata')
+        .then((data) => {
+            if (!data) {
+                return;
+            }
+
+            // For key moments rich results, Google requires clips to be at least 30 seconds long
+            const MIN_DURATION = 30;
+            const defaultThumbnail = 'https://i.vimeocdn.com/portrait/default';
+            const {
+                author,
+                chapters,
+                description: clipDescription,
+                duration: clipDurationSec,
+                embedUrl,
+                thumbsBaseUrl,
+                title: name,
+                uploadDate
+            } = data;
+            const duration = `PT${clipDurationSec}S`;
+            const thumbnailUrl = thumbsBaseUrl ? `${thumbsBaseUrl}_640` : defaultThumbnail;
+            const description = (clipDescription.trim().length) ? clipDescription : `This is "${title}" by ${author} on Vimeo, the home for high quality videos and the people who love them.`;
```

Comment:
If the data is coming from the player, this default value should probably be done there where we are in full control of any updates that may need to be made. This script could very well be copied and never updated on a particular site.

Reply:
@bdougherty are you referring just to `description`. And are you suggesting that we move the logic on line 47 into the player repo? I was debating on whether the description/default should be determined here or in the player repo. I can get on board with moving it.

Reply:
Yes, I think the player itself is more appropriate as it will make things much easier to change if changes are needed in the future. Especially if `getVideoObjectMetadata` is specifically designed for this purpose.

Also was just thinking, does this default description need to be localized?

Reply:
That's a great question. We _do_ have this VideoObject metadata injection already working for on-site videos. And there, [we translate the description](https://github.vimeows.com/Vimeo/vimeo/blob/master/src/Vimeo/CrawlerMarkup.php#L157). I need to look into whether/how we can translate the description on player.

Reply:
Following up: We'll be localizing the default description in the player layer. Pull request [is available here](https://github.vimeows.com/player/player/pull/8450).